kafka-client/src/main/java/dev/responsive/kafka/api/ResponsiveKafkaStreams.java

```diff
@@ -0,0 +1,118 @@
+package dev.responsive.kafka.api;
+
+import static dev.responsive.kafka.config.ResponsiveDriverConfig.CLIENT_ID_CONFIG;
+import static dev.responsive.kafka.config.ResponsiveDriverConfig.CLIENT_SECRET_CONFIG;
+import static dev.responsive.kafka.config.ResponsiveDriverConfig.STORAGE_DATACENTER_CONFIG;
+import static dev.responsive.kafka.config.ResponsiveDriverConfig.STORAGE_HOSTNAME_CONFIG;
+import static dev.responsive.kafka.config.ResponsiveDriverConfig.STORAGE_PORT_CONFIG;
+import static dev.responsive.kafka.config.ResponsiveDriverConfig.TENANT_ID_CONFIG;
+import static dev.responsive.kafka.config.ResponsiveStreamsConfig.verifiedStreamsConfigs;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import dev.responsive.db.CassandraClient;
+import dev.responsive.kafka.clients.ResponsiveKafkaClientSupplier;
+import dev.responsive.kafka.config.ResponsiveDriverConfig;
+import dev.responsive.utils.SessionUtil;
+import java.net.InetSocketAddress;
+import java.time.Duration;
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.common.config.types.Password;
+import org.apache.kafka.streams.KafkaClientSupplier;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.Topology;
+
+public class ResponsiveKafkaStreams extends KafkaStreams {
+
+  private final CqlSession session;
+  private final ScheduledExecutorService executor;
+  private final Admin admin;
+
+  private ResponsiveKafkaStreams(
+      final Topology topology,
+      final StreamsConfig streamsConfigs,
+      final KafkaClientSupplier clientSupplier,
+      final CqlSession session,
+      final Admin admin,
+      final ScheduledExecutorService executor
+  ) {
+    super(topology, streamsConfigs, clientSupplier);
+    this.session = session;
+    this.admin = admin;
+    this.executor = executor;
+  }
+
+  public static ResponsiveKafkaStreams create(
```

Comment:
I don't feel too strongly about this case, just trying to match the previous style in our own code. Happy to align with existing KafkaStreams more closely instead

Reply:
Oh actually this is tricky because you have to call the parent constructor in the very first line of the constructor, and we need to do a few things to prepare the configs before passing them to the underlying `KafkaStreams` constructor. Maybe you can figure out how to make this work, but I was struggling...can I just merge it as-is (with the other fixes) and let you refactor this class the way you want it to look?